### PR TITLE
Add optional random_state kwarg to run_glm()

### PR DESCRIPTION
**Pull request by sjawhar**
Changes proposed in this pull request:
- Add an optional `random_state` kwarg to `nilearn.glm.first_level.first_level.run_glm()`. Because we like reproducibility!

### CODE REVIEW

**Review by bthirion — COMMENTED** [review]
Thx ! 

The change looks good, but shouldn't we propagate the random_state to downstream classes (FirstLevelModel in particular) ?

+ Add a whatsnew entry

**Comment by sjawhar** [comment]
> Thx !
> 
> The change looks good, but shouldn't we propagate the random_state to downstream classes (FirstLevelModel in particular) ?

Done

**Review by bthirion — COMMENTED** [review]
Great ! I'd simply add a whatsnew entry on top of it.

File: nilearn/glm/first_level/first_level.py [hunk 1]
```diff
@@ -133,6 +134,10 @@ def run_glm(Y, X, noise_model='ar1', bins=100, n_jobs=1, verbose=0):
     verbose : int, optional
         The verbosity level. Default=0.
 
+    random_state : int or numpy.random.RandomState, optional
+        Random state seed to sklearn.cluster.KMeans for autoregressive models
+        of order at least 2 ('ar(N)' with n >= 2).
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/first_level.py:139]:
```suggestion
        of order at least 2 ('ar(N)' with n >= 2). Default=None.
        
        .. versionadded:: 0.9.1
                
```

File: nilearn/glm/first_level/first_level.py [hunk 2]
```diff
@@ -322,6 +328,10 @@ class FirstLevelModel(BaseGLM):
         This id will be used to identify a `FirstLevelModel` when passed to
         a `SecondLevelModel` object.
 
+    random_state : int or numpy.random.RandomState, optional
+        Random state seed to sklearn.cluster.KMeans for autoregressive models
+        of order at least 2 ('ar(N)' with n >= 2).
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/first_level.py:333]:
```suggestion
        of order at least 2 ('ar(N)' with n >= 2). Default=None.
        
        .. versionadded:: 0.9.1
        
```

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @sjawhar !
This LGTM, I just have a small suggestion for the docstrings.
Besides, can you add an entry in the change log (`doc/changes/latest.rst`) under the `Enhancements` section, and following the formatting of the other entries?
Note that, since this is your first contribution to Nilearn, you'll need to add your name and link to profile/website to `doc/changes/names.rst`.

**Review by NicolasGensollen — APPROVED** [review]
Thanks @sjawhar !
LGTM!

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Review by ymzayek — COMMENTED** [review]
@sjawhar thanks looks good! Just a small spelling error has to be fixed.

**Pull request merged**